### PR TITLE
wsusserver: Add configure proxy feature

### LINKED ISSUE #7: wsusserver: Ability to use a proxy
Key config params:
`UseProxy`
`ProxyName`
`ProxyServerPort`

Optional params:
`AnonymousProxyAccess`
`ProxyUserName`
`ProxyUserDomain`
`ProxyPassword`
`HasProxyPassword`
`AllowProxyCredentialsOverNonSsl`

### PR DESCRIPTION
**Pull request by jblance**
#### Pull Request (PR) description
Complete feature to configure wsusserver to use a proxy

#### This Pull Request (PR) fixes the following issues
Fixes #7 

#### Task list

- [x] Resource/Class documentation added/updated in README.md?
- [x] Examples appropriately added/updated?
- [ ] Unit tests added/updated?
- [ ] Integration tests added/updated (where possible)?

Testing done on development WSUS server

### CODE REVIEW

**Comment by TraGicCode** [comment]
Let me remove the dsc stuff from the PR template.  I borrowed their layout.

**Comment by TraGicCode** [comment]
Thanks for contributing back.   Do you mind fixing the tests?

**Comment by jblance** [comment]
Tests fixed

**Comment by TraGicCode** [comment]
Hey @jblance ,

Hopefully i will obtain some time to review this soon.  Sorry for the wait and thanks again for the work.

File: spec/classes/config_spec.rb [hunk 1]
```diff
@@ -20,12 +20,72 @@
...
     }
 
-    it {
-      is_expected.to contain_exec('wsus-config-update-languages')
-        .with(
-          logoutput: true,
-          provider: 'powershell',
-        )
-    }
+    it { should contain_exec('wsus-config-update-synchronization').with({
```
Comment (TraGicCode) [inline spec/classes/config_spec.rb:23]: Lets remove the checking of the command and unless attribute content from the test.  This stuff is super hard to maintain and gives way more headache and little value.

**Review by TraGicCode — CHANGES_REQUESTED** [review]
Minor changes to get the tests passing.

**Comment by TraGicCode** [comment]
Also please rebase.

**Comment by TraGicCode** [comment]
Closing since this is an old pr

**Pull request closed without merging**